spt/features/saveloads.cpp

```diff
@@ -0,0 +1,266 @@
+{
+	spt_saveloads.Stop();
+}
+
+ConVar y_spt_hud_saveloads_showcurindex("y_spt_hud_saveloads_showcurindex",
+										"1",
+										0,
+										"Shows the current save/load index of the current save/load process.\n");
+
+namespace patterns
+{
+	PATTERNS(
+		Engine__SignOnState,
+		"OrangeBox-and-older",
+		"80 3D ?? ?? ?? ?? 00 74 06 B8 ?? ?? ?? ?? C3 83 3D ?? ?? ?? ?? 02 B8 ?? ?? ?? ??",
+		"BMS-Retail",
+		"A1 ?? ?? ?? ?? 85 C0 75 ?? B8 ?? ?? ?? ??",
+		"Source-2009-and-Portal-2",
+		"74 ?? 8B 74 87 04 83 7E 18 00 74 2D 8B 0D ?? ?? ?? ?? 8B 49 18");
+};
+
+bool SaveloadsFeature::ShouldLoadFeature() 
+{
+	return true;
+}
+
+void SaveloadsFeature::InitHooks() 
+{
+	FIND_PATTERN_ALL(engine, Engine__SignOnState);
+}
+
+
+void SaveloadsFeature::LoadFeature()
+{
+	if (!spt_afterticks.Works || MATCHES_Engine__SignOnState.empty() || !FrameSignal.Works)
+		return;
+
+	auto match = MATCHES_Engine__SignOnState[0]; 
+	switch (match.ptnIndex)
+	{
+	case 0:
+		ORIG_SignOnState = *(uintptr_t*)(match.ptr + 17);
+		break;
+	case 1:
+		ORIG_SignOnState = *(uintptr_t*)(match.ptr + 1);
+		break;
+	case 2:
+		ORIG_SignOnState = **(uintptr_t**)(match.ptr + 14) + 0x70;
+		break;
+	default: 
+		ORIG_SignOnState = NULL;
+		return;
+	}
+
+	//FinishRestoreSignal.Connect(this, &SaveloadsFeature::FinishRestore);
+	InitCommand(y_spt_saveloads);
+	InitCommand(y_spt_saveloads_stop);
+
+#ifdef SPT_HUD_ENABLED
+	AddHudCallback(
+		"SAVELOADS",
+		[this]()
+		{
+			if (execute)
+				spt_hud.DrawTopHudElement(L"SAVELOADS: %i / %i (%i left)", startIndex, endIndex, endIndex - startIndex);
+		},
+		y_spt_hud_saveloads_showcurindex);
+#endif
+
+	FrameSignal.Connect(this, &SaveloadsFeature::Update);
+}
+
+
+void SaveloadsFeature::Begin(int type_,
+                             const char* segName_,
+                             int startIndex_,
+                             int endIndex_,
+                             int ticksToWait_,
+                             const char* extra)
+{
+
+	if (type_ > 2 || type_ < 0)
+	{
+		Warning("SAVELOADS: Invalid save/load process type! Not continuing.\n");
+		return;
+	}
+
+	if (startIndex_ < 0 || endIndex_ < 0 || startIndex_ > endIndex_)
+	{
+		Warning("SAVELOADS: Invalid start and/or end index.\n");
+		return;
+	}
+
+	this->type = type_;
+	this->prefixName.assign(segName_);
+	this->startIndex = startIndex_;
+	this->endIndex = endIndex_;
+	this->ticksToWait = ticksToWait_;
+	this->extraCommands.assign(extra == nullptr ? "" : extra);
+
+	execute = true;
+
+	Msg(
+		"------\n\nSAVELOADS: Save/load process:\n\
+	- Type: %s\n\
+	- Segment name: \"%s\"\n\
+	- From index %i to index %i (%i save/loads)\n\
+	- Wait time before action: %i\n\
+	- Extra commands: \"%s\"\n\n\
+Please load the save from which save/loading should begin.\n\n------\n",
+	    _saveloadTypes[type_],
+		segName_,
+		startIndex_,
+		endIndex_,
+		endIndex_ - startIndex_ + 1,
+		ticksToWait_,
+		this->extraCommands.c_str());
+
+	EngineConCmd("bind L y_spt_saveloads_stop");
```

Comment:
I thought I commented on this already but this hardcoded bind should probably be deleted

Reply:
You haven't but what should I do instead to communicate that the process can be stopped midway?

Reply:
another option would be to make it a toggle, like +y_spt_saveloads and when you stop holding it you stop saveloading

Reply:
I don't think that's really optimal because many save/loads are long and people leave them running in the background while they alt tab to something else (with y_spt_focus_nosleep)

Reply:
That could be solved via aliases but then again you could also create a +/- alias for this. Either way is fine but the hardcoded bind has gotta go

Reply:
I can't create a +/- for this since it begins once you load a save

Reply:
Fuck it I'll just put up a warning I'm tired of waiting for a reply